lib/libutils/ext/include/util.h

```diff
@@ -50,9 +50,12 @@
 #endif
 
 #ifndef __ASSEMBLER__
-/* Round up the even multiple of size, size has to be a power of 2 */
-#define ROUNDUP(v, size) (((v) + ((__typeof__(v))(size) - 1)) & \
-			  ~((__typeof__(v))(size) - 1))
+/* Round up the even multiple of size */
+#define ROUNDUP(v, size) \
+	(IS_POWER_OF_TWO(size) ? \
```

Comment:
How about using two helper macros, one for the power of 2 case and one for the other?
While you're at it, it would be nice to evaluate the arguments only once, like in `UDIV_ROUND_NEAREST()` below.
Same below for `ROUNDDOWN()`.
Would it be possible to fix `ROUNDUP_OVERFLOW()` to allow a size that isn't a power of 2? It would align better with the new versions of `ROUNDUP()` and `ROUNDDOWN()`

Reply:
> How about using two helper macros, one for the power of 2 case and one for the other?

Okay, i'll go for `ROUNDUP_POW2()` and `ROUNDUP()` if that sounds ok to you (@jforissier ?)

That said, how to ensure one does not use a non power of 2 on `ROUNDUP_POW2()` for example. We cannot use `assert()` because the macro is used in `uint8_t buf[ROUNDUP(foo, bar)]` like array definitions?
One way: use non-pow2 `ROUNDUP()` macro in these cases (we expect the result is evaluated at build time).
Another way: have `ROUNDUP_POW2()` and `ROUNDDOWN_POW2()` to get the bit shift value instead of the power-of-2 size value.

> While you're at it, it would be nice to evaluate the arguments only once, like in UDIV_ROUND_NEAREST() below.
Same below for ROUNDDOWN().

Ok, i'll see that.

> Would it be possible to fix ROUNDUP_OVERFLOW() to allow a size that isn't a power of 2? It would align better with the new versions of ROUNDUP() and ROUNDDOWN()

I agree. I'll do that.

Reply:
Ok, i'll update this P-R